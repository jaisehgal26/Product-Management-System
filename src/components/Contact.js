import React, { Suspense } from "react";
import { Formik, Form, Field, ErrorMessage, useFormikContext } from "formik";
import * as Yup from "yup";
import Axios from "axios";
import Navbar from "./Navbar";
import { Prompt } from "react-router-dom";
import Footer from "./Footer";
import Fade from "react-reveal/Fade";

export default function Contact() {
  const initialValues = { name: "", email: "", subject: "", message: "" };
  const onSubmit = (value, { resetForm }) => {
    Axios.post("http://localhost:3000/contact", {
      name: value.name,
      email: value.email,
      subject: value.subject,
      message: value.message,
    });
    resetForm({ value: "" });
  };
  const PromptIfDirty = () => {
    const formik = useFormikContext();
    return (
      <Prompt
        when={formik.dirty && formik.submitCount === 0}
        message="Are you sure you want to leave? You have with unsaved changes."
      />
    );
  };
  const validationSchema = Yup.object({
    name: Yup.string().required("Required"),
    email: Yup.string().email("Enter valid Email").required("Required"),
    subject: Yup.string().required("Required"),
    message: Yup.string().required("Required"),
  });
  return (
    <>
      <Navbar />

      <Suspense
        fallback={
          <div className="center spinner-grow text-warning" role="status">
            <span className="sr-only">Loading...</span>
          </div>
        }
      >
        <Fade bottom>
          <div className="container-xxl py-6">
            <div className="container">
              <div
                className="text-center mx-auto mb-5 "
                style={{ maxwidth: "500px" }}
              >
                <p className="text-primary text-uppercase mb-2"># Contact Us</p>
                <h1 className="display-6 mb-4 Playfair">
                  If You Have Any Query, Please Contact Us
                </h1>
              </div>
              <div className="row g-0 justify-content-center">
                <div className="col-lg-8 ">
                  <Formik
                    initialValues={initialValues}
                    validationSchema={validationSchema}
                    onSubmit={onSubmit}
                  >
                    <Form>
                      <PromptIfDirty />
                      <div className="row g-3">
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="name"
                              name="name"
                              className="form-control"
                            />
                            <label htmlFor="name" className="form-label">
                              Name
                            </label>
                            <ErrorMessage name="name" />
                          </div>
                        </div>
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="email"
                              name="email"
                              className="form-control"
                            />
                            <label htmlFor="email" className="form-label">
                              Email
                            </label>
                            <ErrorMessage name="email" />
                          </div>
                        </div>
                        <div className="col-12">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="subject"
                              name="subject"
                              className="form-control"
                            />
                            <label htmlFor="subject" className="form-label">
                              Subject
                            </label>
                            <ErrorMessage name="subject" />
                          </div>
                        </div>
                        <div className="col-12">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="message"
                              name="message"
                              className="form-control"
                              style={{ height: "5rem" }}
                            />
                            <label htmlFor="message" className="form-label">
                              Message
                            </label>
                            <ErrorMessage name="message" />
                          </div>
                        </div>
                        <div className="col-12 text-center">
                          <button
                            className="btn btn-primary rounded-pill py-3 px-5"
                            type="submit"
                          >
                            Send Message
                          </button>
                        </div>
                      </div>
                    </Form>
                  </Formik>
                </div>
              </div>
            </div>
          </div>
        </Fade>
      </Suspense>
      <Footer />
    </>
  );
}
