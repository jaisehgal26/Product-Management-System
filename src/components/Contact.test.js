import * as React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Contact from "./Contact";

test("should render correctly", () => {
  const wrapper = toJson(shallow(<Contact />));
  expect(wrapper).toMatchSnapshot();
});
