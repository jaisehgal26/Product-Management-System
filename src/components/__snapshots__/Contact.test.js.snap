// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Fragment>
  <Navbar />
  <Suspense
    fallback={
      <div
        className="center spinner-grow text-warning"
        role="status"
      >
        <span
          className="sr-only"
        >
          Loading...
        </span>
      </div>
    }
  >
    <Fade
      bottom={true}
    >
      <div
        className="container-xxl py-6"
      >
        <div
          className="container"
        >
          <div
            className="text-center mx-auto mb-5 "
            style={
              Object {
                "maxwidth": "500px",
              }
            }
          >
            <p
              className="text-primary text-uppercase mb-2"
            >
              # Contact Us
            </p>
            <h1
              className="display-6 mb-4 Playfair"
            >
              If You Have Any Query, Please Contact Us
            </h1>
          </div>
          <div
            className="row g-0 justify-content-center"
          >
            <div
              className="col-lg-8 "
            >
              <Formik
                initialValues={
                  Object {
                    "email": "",
                    "message": "",
                    "name": "",
                    "subject": "",
                  }
                }
                onSubmit={[Function]}
                validationSchema={
                  ObjectSchema {
                    "_blacklist": ReferenceSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_excludedEdges": Array [],
                    "_mutate": undefined,
                    "_nodes": Array [
                      "message",
                      "subject",
                      "email",
                      "name",
                    ],
                    "_sortErrors": [Function],
                    "_typeError": [Function],
                    "_whitelist": ReferenceSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "conditions": Array [],
                    "deps": Array [],
                    "exclusiveTests": Object {},
                    "fields": Object {
                      "email": StringSchema {
                        "_blacklist": ReferenceSet {
                          "list": Set {},
                          "refs": Map {},
                        },
                        "_blacklistError": undefined,
                        "_mutate": undefined,
                        "_typeError": [Function],
                        "_whitelist": ReferenceSet {
                          "list": Set {},
                          "refs": Map {},
                        },
                        "_whitelistError": undefined,
                        "conditions": Array [],
                        "deps": Array [],
                        "exclusiveTests": Object {
                          "email": false,
                          "required": true,
                        },
                        "spec": Object {
                          "abortEarly": true,
                          "nullable": false,
                          "presence": "required",
                          "recursive": true,
                          "strict": false,
                          "strip": false,
                        },
                        "tests": Array [
                          [Function],
                          [Function],
                        ],
                        "transforms": Array [
                          [Function],
                        ],
                        "type": "string",
                      },
                      "message": StringSchema {
                        "_blacklist": ReferenceSet {
                          "list": Set {},
                          "refs": Map {},
                        },
                        "_blacklistError": undefined,
                        "_mutate": undefined,
                        "_typeError": [Function],
                        "_whitelist": ReferenceSet {
                          "list": Set {},
                          "refs": Map {},
                        },
                        "_whitelistError": undefined,
                        "conditions": Array [],
                        "deps": Array [],
                        "exclusiveTests": Object {
                          "required": true,
                        },
                        "spec": Object {
                          "abortEarly": true,
                          "nullable": false,
                          "presence": "required",
                          "recursive": true,
                          "strict": false,
                          "strip": false,
                        },
                        "tests": Array [
                          [Function],
                        ],
                        "transforms": Array [
                          [Function],
                        ],
                        "type": "string",
                      },
                      "name": StringSchema {
                        "_blacklist": ReferenceSet {
                          "list": Set {},
                          "refs": Map {},
                        },
                        "_blacklistError": undefined,
                        "_mutate": undefined,
                        "_typeError": [Function],
                        "_whitelist": ReferenceSet {
                          "list": Set {},
                          "refs": Map {},
                        },
                        "_whitelistError": undefined,
                        "conditions": Array [],
                        "deps": Array [],
                        "exclusiveTests": Object {
                          "required": true,
                        },
                        "spec": Object {
                          "abortEarly": true,
                          "nullable": false,
                          "presence": "required",
                          "recursive": true,
                          "strict": false,
                          "strip": false,
                        },
                        "tests": Array [
                          [Function],
                        ],
                        "transforms": Array [
                          [Function],
                        ],
                        "type": "string",
                      },
                      "subject": StringSchema {
                        "_blacklist": ReferenceSet {
                          "list": Set {},
                          "refs": Map {},
                        },
                        "_blacklistError": undefined,
                        "_mutate": undefined,
                        "_typeError": [Function],
                        "_whitelist": ReferenceSet {
                          "list": Set {},
                          "refs": Map {},
                        },
                        "_whitelistError": undefined,
                        "conditions": Array [],
                        "deps": Array [],
                        "exclusiveTests": Object {
                          "required": true,
                        },
                        "spec": Object {
                          "abortEarly": true,
                          "nullable": false,
                          "presence": "required",
                          "recursive": true,
                          "strict": false,
                          "strip": false,
                        },
                        "tests": Array [
                          [Function],
                        ],
                        "transforms": Array [
                          [Function],
                        ],
                        "type": "string",
                      },
                    },
                    "spec": Object {
                      "abortEarly": true,
                      "nullable": false,
                      "presence": "optional",
                      "recursive": true,
                      "strict": false,
                      "strip": false,
                    },
                    "tests": Array [],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "object",
                  }
                }
              >
                <Form>
                  <PromptIfDirty />
                  <div
                    className="row g-3"
                  >
                    <div
                      className="col-md-6"
                    >
                      <div
                        className="form-floating"
                      >
                        <Field
                          className="form-control"
                          id="name"
                          name="name"
                          type="text"
                        />
                        <label
                          className="form-label"
                          htmlFor="name"
                        >
                          Name
                        </label>
                        <FormikConnect(ErrorMessageImpl)
                          name="name"
                        />
                      </div>
                    </div>
                    <div
                      className="col-md-6"
                    >
                      <div
                        className="form-floating"
                      >
                        <Field
                          className="form-control"
                          id="email"
                          name="email"
                          type="text"
                        />
                        <label
                          className="form-label"
                          htmlFor="email"
                        >
                          Email
                        </label>
                        <FormikConnect(ErrorMessageImpl)
                          name="email"
                        />
                      </div>
                    </div>
                    <div
                      className="col-12"
                    >
                      <div
                        className="form-floating"
                      >
                        <Field
                          className="form-control"
                          id="subject"
                          name="subject"
                          type="text"
                        />
                        <label
                          className="form-label"
                          htmlFor="subject"
                        >
                          Subject
                        </label>
                        <FormikConnect(ErrorMessageImpl)
                          name="subject"
                        />
                      </div>
                    </div>
                    <div
                      className="col-12"
                    >
                      <div
                        className="form-floating"
                      >
                        <Field
                          className="form-control"
                          id="message"
                          name="message"
                          style={
                            Object {
                              "height": "5rem",
                            }
                          }
                          type="text"
                        />
                        <label
                          className="form-label"
                          htmlFor="message"
                        >
                          Message
                        </label>
                        <FormikConnect(ErrorMessageImpl)
                          name="message"
                        />
                      </div>
                    </div>
                    <div
                      className="col-12 text-center"
                    >
                      <button
                        className="btn btn-primary rounded-pill py-3 px-5"
                        type="submit"
                      >
                        Send Message
                      </button>
                    </div>
                  </div>
                </Form>
              </Formik>
            </div>
          </div>
        </div>
      </div>
    </Fade>
  </Suspense>
  <Footer />
</Fragment>
`;
